server/matcher/match.go

```diff
@@ -102,23 +106,29 @@ func CheckMarketBuyBuffer(book Booker, ord *order.MarketOrder, marketBuyBuffer f
 	return ord.Remaining() >= BaseToQuote(book.BestSell().Rate, minBaseAsset)
 }
 
+// OrderRevealed combines an Order interface with a Preimage.
+type OrderRevealed struct {
+	Order order.Order // Do not embed so OrderRevealed is not an order.Order.
+	order.Preimage
+}
```

Comment:
That's pretty much right, and I started with it embedded and lots of hard to track bugs manifested.  It's clearly safer not to have the possibility in the future of such bugs unless we really want to have this as an `order.Order` for other reasons. Explicitly hitting the `Order` field did make things more bullet proof.  I think this is OK because there are very limited areas where OrderRevealed is used.
We could make `Preimage` a field rather than embedding.  I'll go ahead with that.